CONTRIBUTING.rst

```diff
@@ -4,74 +4,11 @@
 Contributing
 ============
 
-The easiest way to let us know about your awesome work is to send a pull
-request on GitHub or in IRC.
+Before contributing code, please note:
 
-Use **summary (bug #bugzilla bug ID#)** for the commit message. Eg::
+* You agree to license your contributions under the [license](https://github.com/mozilla/olympia/blob/master/LICENSE).
+* Ask on the [dev-addons mailing list](https://mail.mozilla.org/listinfo/dev-addons) or in GitHub before making changes.
+* Follow [PEP8](http://www.python.org/dev/peps/pep-0008/), [jshint](http://www.jshint.com/) and our other [style guide conventions](http://mozweb.readthedocs.org/en/latest/index.html).
+* Please write tests and read the docs on [olympia](http://olympia.readthedocs.org/en/latest/).
 
-    Improve contributing instructions (bug 1041234)
-
-Point us to a branch with your new code and we'll
-go from there.  You can attach a patch to a bug if you're more comfortable that
-way.
-
-Please read the :ref:`style`.
-
-The Perfect Git Configuration
```

Comment:
I never liked this configuration anyway. Although I think this is in the docs somewhere else too. I prefer something like https://help.github.com/articles/configuring-a-remote-for-a-fork/ with `master` setup to track `upstream`. Since you asked (you obviously didn't).